installers/docker-compose/scripts/setup.sh

```diff
@@ -0,0 +1,144 @@
+# e.g. http://169.254.169.254/latest/meta-data/public-hostname
+# then there is not a lot to worry about because inside each Docker container
+# this host can be referred as-is
+#
+# However, if we use `localhost`, we cannot use refer to that hostname cross containers
+# even if we expose the ports
+# See https://pythonspeed.com/articles/docker-connection-refused/
+# 
+# However, we can refer to other container using their container name
+# thanks to Docker Compose `networks` setting
+# e.g. mongodb://mongod:27017 is accessible from containers other than `mongod`, but not mongodb://localhost:27017
+# See https://docs.docker.com/compose/networking/
+#
+# But if we use host name like `localhost` or `gitlab`, or container's name, or IP address
+# Java processes might require us to also provide Subject Alternative Name (SAN)
+# We do this by using extensions in V3 of OpenSSL config
+# This is important in order to avoid errors such as `No name matching` or `No subject alternative names matching IP address`
+# when Java calling to these addresses
+# See https://medium.com/@antelle/how-to-generate-a-self-signed-ssl-certificate-for-an-ip-address-f0dd8dddf754
+
+##########################################
+# 	Clean up and reset build directory
+##########################################
+
+[ -e $BUILD_DIR ] && rm -r $BUILD_DIR
+mkdir -p $BUILD_DIR
+mkdir -p $BUILD_DIR/ssl
+mkdir -p $BUILD_DIR/container-data
+mkdir -p $BUILD_DIR/generated-configs
+mkdir -p $BUILD_DIR/generated-configs/gitlab
+mkdir -p $BUILD_DIR/generated-configs/sdlc
+mkdir -p $BUILD_DIR/generated-configs/engine
+mkdir -p $BUILD_DIR/generated-configs/studio
+
+
+##########################################
+# 	Generate certificates and secrets
+##########################################
+
+HOST_KEY_FILE=$BUILD_DIR/ssl/$HOST_DNS_NAME".key"
+HOST_CERT_FILE=$BUILD_DIR/ssl/$HOST_DNS_NAME".crt"
+HOST_DER_FILE=$BUILD_DIR/ssl/$HOST_DNS_NAME".der"
+
+GITLAB_KEY_FILE=/etc/gitlab/ssl/$HOST_DNS_NAME".key"
+GITLAB_CERT_FILE=/etc/gitlab/ssl/$HOST_DNS_NAME".crt"
+TRUST_STORE=$BUILD_DIR/ssl/truststore.jks
+
+# Generate secrets and passwords
+MONGO_PASSWORD=$(openssl rand -base64 8 | sed 's:/::g')
+GITLAB_ROOT_PASSWORD=$(openssl rand -base64 8 | sed 's:/::g')
+GITLAB_PRIVATE_ACCESS_TOKEN=$(openssl rand -base64 8 | sed 's:/::g')
+
+# Generate and convert certificate
+# See https://www.digitalocean.com/community/tutorials/openssl-essentials-working-with-ssl-certificates-private-keys-and-csrs
+
+GITLAB_SERVICE_NAME=$(echo $(grep -v '^#' $PWD/../templates/.env | grep -e "GITLAB_SERVICE_NAME" | sed -e 's/.*=//'))
+
+CERT_CONFIG_FILE=$BUILD_DIR/cert.cnf
+[ -e $CERT_CONFIG_FILE ] && rm $CERT_CONFIG_FILE
+cp -r $PWD/../templates/cert.cnf $CERT_CONFIG_FILE
+sed -i 's/__HOST_DNS_NAME__/'$HOST_DNS_NAME'/g' $CERT_CONFIG_FILE
+sed -i 's/__GITLAB_SERVICE_NAME__/'$GITLAB_SERVICE_NAME'/g' $CERT_CONFIG_FILE
+sed -i 's/__HOST_DNS__/'$HOST_DNS'/g' $CERT_CONFIG_FILE
+
+openssl req \
```

Comment:
I'm running into this on EC2. https://security.stackexchange.com/questions/239369/openssl-general-namebad-ip-address-set-in-alt-names

Reply:
addressed